components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/definitions/APIDefinitionFromSwagger12.java

```diff
@@ -0,0 +1,418 @@
+        JSONParser parser = new JSONParser();
+        JSONObject resourceConfigs;
+        Set<URITemplate> uriTemplates = new LinkedHashSet<URITemplate>();
+        try {
+            resourceConfigs = (JSONObject) parser.parse(resourceConfigsJSON);
+            JSONArray resources = (JSONArray) resourceConfigs.get("resources");
+            //Iterating each resourcePath config
+            for (Object resource : resources) {
+                JSONObject resourceConfig = (JSONObject) resource;
+                APIManagerConfiguration config = ServiceReferenceHolder.getInstance().getAPIManagerConfigurationService().getAPIManagerConfiguration();
+
+                Map<String, Environment> environments = config.getApiGatewayEnvironments();
+                Environment environment = null;
+                String endpoint = null;
+                if (environments != null) {
+                    Set<String> publishedEnvironments = api.getEnvironments();
+                    if (publishedEnvironments.isEmpty() || publishedEnvironments.contains("none")) {
+                        environment = environments.values()
+                                .toArray(new Environment[environments.size()])[0];
+                        String gatewayEndpoint = environment.getApiGatewayEndpoint();
+                        if (gatewayEndpoint.contains(",")) {
+                            endpoint = gatewayEndpoint.split(",")[0];
+                        } else {
+                            endpoint = gatewayEndpoint;
+                        }
+                    } else {
+                        for (String environmentName : publishedEnvironments) {
+                            // find environment that has hybrid type
+                            if (APIConstants.GATEWAY_ENV_TYPE_HYBRID
+                                    .equals(environments.get(environmentName).getType())) {
+                                environment = environments.get(environmentName);
+                                break;
+                            }
+                        }
+                        //if not having any hybrid environment give 1st environment in api published list
+                        if (environment == null) {
+                            environment = environments.get(publishedEnvironments.toArray()[0]);
+                        }
+                        String gatewayEndpoint = environment.getApiGatewayEndpoint();
+                        if (gatewayEndpoint != null && gatewayEndpoint.contains(",")) {
+                            endpoint = gatewayEndpoint.split(",")[0];
+                        } else {
+                            endpoint = gatewayEndpoint;
+                        }
+                    }
+                }
+                //String apiPath = APIUtil.getAPIPath(apiIdentifier);
+                if (endpoint.endsWith(RegistryConstants.PATH_SEPARATOR)) {
+                    endpoint.substring(0, endpoint.length() - 1);
```

Comment:
We're not assigning the substring'ed value to any variable. So this operation's outcome hasn't been used anywhere.